highfidelity/hifi
# Add grid entities to create

https://highfidelity.manuscript.com/f/cases/20171

Pre-Merge Test Plan:
- Enter Interface in desktop mode into any domain with edit permissions
- Enter the following line into Developer menu > Scripting > Console and hit Enter:
Entities.addEntity({type: "Grid", position: MyAvatar.position})
- Refresh your Entity List to verify a Grid entity is created as expected
- In the Entity List, open the Types dropdown, verify Grid is the last type choice with a Grid icon next to it, and verify that it filters Grid entities in and out of the list as expected
- Select the Grid entity in the list and then view the Properties tab
- Verify the same Grid icon appears at the top left of the Properties tab
- Verify Grid entity properties contain the Spatial, Behavior, and Physics properties groups but not the Collision property gorup
- Verify the Grid-specific properties that appear at the top under Parent Joint Index are:
Color - defaults to white
Follow Camera - checkbox defaults to checked
Major Grid Every - number defaults to 5
Minor Grid Every - number defaults to 1.00
- Uncheck Follow Camera (it's behavior is not working as intended and will be bugged)
- Expand the Grid entity dimensions to be large (at least 50 X and 50 Y dimensions). It is expected that Z dimensions cannot change.
- Verify changing the Color property changes the grid line colors
- Verify that Minor Grid Every changes the distance between grid lines (enter numbers manually such as 0.5, 1.0, 2.0, 5.0 to see the changes instead of dragging)
- Verify that Major Grid Every changes the interval of smaller grid lines at which a thicker line is drawn (use the step arrows instead of dragging to see the changes more closely)
